src/snowflake/snowpark/session.py

```diff
@@ -1568,6 +1568,65 @@ def query_tag(self, tag: str) -> None:
             self._conn.run_query("alter session unset query_tag")
         self._query_tag = tag
 
+    def append_query_tag(self, tag: Union[str, dict], seperator: str = ",") -> None:
```

Comment:
As far as I understand people already use json encoded strings with the query tag property. Adding support for automatic json serialization could hurt since that's what I've done for appending. Do you think that should be part of this change or added separately since it affects the current api?

Reply:
No, please file a follow-up ticket.
CC @sfc-gh-achandrasekaran for prioritization